### PR TITLE
New Google Workspace Rule to Monitor When Admin Provisions Custom Role

### PR DESCRIPTION
**Pull request by andrea-youwakim**
adding newly created rule to monitor when google workspace admin provisions a custom role

### Background

<High level overview here>

### Changes

* <Describe changes here>

### Testing

* 
Screen Shot 2022-12-14 at 2 38 50 PM

### CODE REVIEW

File: rules/gsuite_activityevent_rules/google_workspace_admin_custom_role.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+AnalysisType: rule
+Description: A google workspace administrator provisioned a custom role to another user.
```
Comment (jpl5280) [inline rules/gsuite_activityevent_rules/google_workspace_admin_custom_role.yml:2]:
```suggestion
Description: A Google Workspace administrator created a new custom administrator role.
```

File: rules/gsuite_activityevent_rules/google_workspace_admin_custom_role.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+
+def title(event):
+    # (Optional) Return a string which will be shown as the alert title.
+    # If no 'dedup' function is defined, the return value of this method
+    # will act as deduplication string.
+    return (
+        f"Google Workspace Administrator "
+        f"[{event.get('actor',{}).get('email','no-email-provided')}] "
+        f"provisioned the role "
```
Comment (jpl5280) [inline rules/gsuite_activityevent_rules/google_workspace_admin_custom_role.py:17]:
```suggestion
        f"created a new admin role "
```

File: rules/gsuite_activityevent_rules/google_workspace_admin_custom_role.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        f"[{event.get('parameters',{}).get('ROLE_NAME','no-role-name-provided')}] "
+        f" from ip address [{event.get('ipAddress')}]."
```
Comment (jpl5280) [inline rules/gsuite_activityevent_rules/google_workspace_admin_custom_role.py:19]: id remove this part (more appropriate for alert context)

**Review by jpl5280 — CHANGES_REQUESTED** [review]

File: rules/gsuite_activityevent_rules/google_workspace_admin_custom_role.py (hunk 3 above)
Comment (andrea-youwakim) [reply rules/gsuite_activityevent_rules/google_workspace_admin_custom_role.py:19]: hmm, i'm not sure what you're looking for me to remove, could you be more explicit about this comment?

Comment (andrea-youwakim) [reply rules/gsuite_activityevent_rules/google_workspace_admin_custom_role.py:19]: @jpl5280 for notif purposes :)

File: packs/gsuite_reports.yml [hunk 4]
```diff
@@ -21,6 +21,7 @@ PackDefinition:
     - GSuite.SuspiciousLogins
     - GSuite.TwoStepVerification
     - GSuite.UserSuspended
+    - Google.Workspace.Admin.Custom.Role
```
Comment (edyesed) [inline packs/gsuite_reports.yml:24]: well, this pack's `Description:` says that it is GSuite Reports based, and the contents look to be more generally GSuite. Can you please update the Description to be `Panther GSuite Detections`?  I can cut a PR to bring in missing detections to the pack

**Review by edyesed — COMMENTED** [review]
I think an update to pack Description is warranted.

File: rules/gsuite_activityevent_rules/google_workspace_admin_custom_role.py (hunk 3 above)
Comment (jpl5280) [reply rules/gsuite_activityevent_rules/google_workspace_admin_custom_role.py:19]: Ah, was suggesting you remove the whole line `f" from ip address [{event.get('ipAddress')}]."`

File: packs/gsuite_reports.yml (hunk 4 above)
Comment (andrea-youwakim) [reply packs/gsuite_reports.yml:24]: changing in next commit!

File: rules/gsuite_activityevent_rules/google_workspace_admin_custom_role.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    )
+
+
+def title(event):
+    # (Optional) Return a string which will be shown as the alert title.
+    # If no 'dedup' function is defined, the return value of this method
+    # will act as deduplication string.
+    return (
+        f"Google Workspace Administrator "
+        f"[{event.get('actor',{}).get('email','no-email-provided')}] "
```
Comment (jpl5280) [inline rules/gsuite_activityevent_rules/google_workspace_admin_custom_role.py:16]: @edyesed - whats our best practice around non-existing values? I've seen <> used a few times?

Comment (edyesed) [reply rules/gsuite_activityevent_rules/google_workspace_admin_custom_role.py:16]: it's a bit in flux, recent practice has been `<NO_WHATEVER_THING>` with the `<>` and all caps.

Comment (andrea-youwakim) [reply rules/gsuite_activityevent_rules/google_workspace_admin_custom_role.py]: just made this change!

**Review by edyesed — APPROVED** [review]
Adds a new GSuite Detection for when GSuite Admins create a custom role with Admin privileges.

**Comment by andrea-youwakim** [comment]
hi @jpl5280 do you mind taking another peek at this ? :) 

thanks in advance!

**Review by jpl5280 — APPROVED** [review]
LG. Eventually we need to migrate all of these to Google Workspace vs. GSuite (its not called that anymore). But a task for another day :)

**Pull request merged**